test/test_synthetic.py

```diff
@@ -113,45 +113,11 @@ def length_of_ray(target_xyz:list, model_zs, los):
 
     # iterate over height levels
     for hh, ht in enumerate(hgt_lvls):
-
         llh = [xx, yy, np.full(yy.shape, ht)]
-
         xyz = np.stack(lla2ecef(llh[1], llh[0], np.full(yy.shape, ht)), axis=-1)
         LOS = los.getLookVectors(ht, llh, xyz, yy)
-
-        cos_factor = None
-
-        # 2d array where output is added to; one per hgt lvl
-        outSubs = outputArrs[hh, ...]
-        # iterate over all model levels
-        for zz in range(model_zs.size-1):
-            low_ht = model_zs[zz]
-            high_ht = model_zs[zz + 1]
-
-            if (high_ht <= ht) or (low_ht >= 50000):
-                continue
-
-            # If high_ht > max_tropo_height - integral only up to max tropo
-            if high_ht > 50000:
-                high_ht = 50000
-
-            # If low_ht < height of point - integral only up to height of point
-            if low_ht < ht:
-                low_ht = ht
-
-            # Continue only if needed - 1m troposphere does nothing
-            if np.abs(high_ht - low_ht) < 1.0:
-                continue
-
-            low_xyz = getTopOfAtmosphere(xyz, LOS, low_ht, factor=cos_factor)
-            high_xyz = getTopOfAtmosphere(xyz, LOS, high_ht, factor=cos_factor)
-
-            ray_length = np.linalg.norm(high_xyz - low_xyz, axis=-1)
-
-            outSubs   += ray_length
-
-            cos_factor = (high_ht - low_ht) / ray_length if cos_factor is None else cos_factor
-
+        ray_lengths = build_ray(model_zs, ht, xyz, LOS)[0]
```

Comment:
My main comment here is that as part of testing you are ignoring the optional arguments to the function. You should test them...

Reply:
Agreed, but suggest we do it in another PR